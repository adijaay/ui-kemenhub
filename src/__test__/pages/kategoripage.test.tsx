import KategoriPage from "@/pages/kategori";
import "@testing-library/jest-dom";
import { render, waitFor } from "@testing-library/react";
import { useRouter } from "next/router";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

jest.mock("@/components/ui/Page", () => {
  return jest.fn().mockImplementation(({ children }) => {
    return <div>{children}</div>;
  });
});

jest.mock("@/utils/firebase", () => ({
  analytics: jest.fn(),
  logEvent: jest.fn(),
}));

describe("KategoriPage", () => {
  (useRouter as jest.Mock).mockReturnValue({
    query: {},
  });

  it("render kategori page correctly", async () => {
    const page = render(<KategoriPage />);
    await waitFor(() => {
      expect(page).toMatchSnapshot();
    });
  });
});
