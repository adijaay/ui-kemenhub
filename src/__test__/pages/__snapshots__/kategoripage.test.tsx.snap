// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KategoriPage render kategori page correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
        >
          <p
            class="font-semibold text-[#1f1f1f] undefined"
          >
            Kategori Penerima PIP
          </p>
          <ul
            class="flex list-none flex-col gap-2"
          >
            <li
              class="flex gap-2 text-sm font-semibold text-[#1f1f1f]"
            >
              <div
                class="mt-[6px] h-1 w-1 shrink-0 rounded-full bg-[#182230]"
              />
              <span>
                Peserta Pemegang PIP
              </span>
            </li>
            <li
              class="flex gap-2 text-sm font-semibold text-[#1f1f1f]"
            >
              <div
                class="mt-[6px] h-1 w-1 shrink-0 rounded-full bg-[#182230]"
              />
              <span>
                Peserta Didik dari keluarga miskin/rentan miskin dan/atau dengan pertimbangan khusus:
              </span>
            </li>
            <ul
              class="flex list-none flex-col gap-2"
            >
              <li
                class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
              >
                <div
                  class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
                />
                <span>
                  Peserta Didik dari keluarga peserta Program Keluarga Harapan
                </span>
              </li>
              <li
                class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
              >
                <div
                  class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
                />
                <span>
                  Peserta Didik dari keluarga pemegang KartuKeluarga Sejahtera
                </span>
              </li>
              <li
                class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
              >
                <div
                  class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
                />
                <span>
                  Peserta Didik yang berstatus yatimpiatu/yatim/piatu dari sekolah/panti sosial/panti asuhan
                </span>
              </li>
              <li
                class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
              >
                <div
                  class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
                />
                <span>
                  Peserta Didik yang terkena dampak bencana alam
                </span>
              </li>
              <li
                class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
              >
                <div
                  class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
                />
                <span>
                  Peserta Didik yang tidak bersekolah (drop out) yang diharapkan kembali bersekolah
                </span>
              </li>
              <li
                class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
              >
                <div
                  class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
                />
                <span>
                  Peserta Didik yang mengalami kelainan fisik, korban musibah, dari orang tua yang mengalami pemutusan hubungan kerja, di daerah konflik, dari keluarga terpidana, berada di Lembaga Pemasyarakatan, memiliki lebih dari 3 (tiga) saudara yang tinggal serumah
                </span>
              </li>
              <li
                class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
              >
                <div
                  class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
                />
                <span>
                  Peserta pada lembaga kursus atau satuan pendidikan nonformal lainnya
                </span>
              </li>
            </ul>
          </ul>
        </div>
        <button
          class="relative flex items-center justify-between overflow-hidden rounded-xl bg-[#AF2A2D] px-4 py-3"
          type="button"
        >
          <img
            alt="vector-1"
            class="absolute right-0 top-13 w-[100px] rounded-br-xl"
            src="/decorations/vector-button-1.png"
          />
          <img
            alt="vector-2"
            class="absolute bottom-[27px] right-[35px] w-[100px] rounded-br-xl"
            src="/decorations/vector-button-2.png"
          />
          <img
            alt="vector-3"
            class="absolute left-[0px] top-[28px] w-[100px] rounded-bl-xl"
            src="/decorations/vector-button-3.png"
          />
          <div
            class="z-10 w-full text-start"
          >
            <p
              class="text-xs font-normal leading-[18px] text-white"
            >
              Baca Juga
            </p>
            <p
              class="text-sm font-bold leading-5 text-white"
            >
              Tujuan Program Indonesia Pintar
            </p>
          </div>
          <svg
            class="tabler-icon tabler-icon-chevron-right "
            fill="none"
            height="24"
            stroke="#ffffff"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
      >
        <p
          class="font-semibold text-[#1f1f1f] undefined"
        >
          Kategori Penerima PIP
        </p>
        <ul
          class="flex list-none flex-col gap-2"
        >
          <li
            class="flex gap-2 text-sm font-semibold text-[#1f1f1f]"
          >
            <div
              class="mt-[6px] h-1 w-1 shrink-0 rounded-full bg-[#182230]"
            />
            <span>
              Peserta Pemegang PIP
            </span>
          </li>
          <li
            class="flex gap-2 text-sm font-semibold text-[#1f1f1f]"
          >
            <div
              class="mt-[6px] h-1 w-1 shrink-0 rounded-full bg-[#182230]"
            />
            <span>
              Peserta Didik dari keluarga miskin/rentan miskin dan/atau dengan pertimbangan khusus:
            </span>
          </li>
          <ul
            class="flex list-none flex-col gap-2"
          >
            <li
              class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
            >
              <div
                class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
              />
              <span>
                Peserta Didik dari keluarga peserta Program Keluarga Harapan
              </span>
            </li>
            <li
              class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
            >
              <div
                class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
              />
              <span>
                Peserta Didik dari keluarga pemegang KartuKeluarga Sejahtera
              </span>
            </li>
            <li
              class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
            >
              <div
                class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
              />
              <span>
                Peserta Didik yang berstatus yatimpiatu/yatim/piatu dari sekolah/panti sosial/panti asuhan
              </span>
            </li>
            <li
              class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
            >
              <div
                class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
              />
              <span>
                Peserta Didik yang terkena dampak bencana alam
              </span>
            </li>
            <li
              class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
            >
              <div
                class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
              />
              <span>
                Peserta Didik yang tidak bersekolah (drop out) yang diharapkan kembali bersekolah
              </span>
            </li>
            <li
              class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
            >
              <div
                class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
              />
              <span>
                Peserta Didik yang mengalami kelainan fisik, korban musibah, dari orang tua yang mengalami pemutusan hubungan kerja, di daerah konflik, dari keluarga terpidana, berada di Lembaga Pemasyarakatan, memiliki lebih dari 3 (tiga) saudara yang tinggal serumah
              </span>
            </li>
            <li
              class="flex gap-2 pl-4 text-sm font-normal text-[#1f1f1f]"
            >
              <div
                class="mt-[6px] h-1 w-1 shrink-0 rounded-full border border-black"
              />
              <span>
                Peserta pada lembaga kursus atau satuan pendidikan nonformal lainnya
              </span>
            </li>
          </ul>
        </ul>
      </div>
      <button
        class="relative flex items-center justify-between overflow-hidden rounded-xl bg-[#AF2A2D] px-4 py-3"
        type="button"
      >
        <img
          alt="vector-1"
          class="absolute right-0 top-13 w-[100px] rounded-br-xl"
          src="/decorations/vector-button-1.png"
        />
        <img
          alt="vector-2"
          class="absolute bottom-[27px] right-[35px] w-[100px] rounded-br-xl"
          src="/decorations/vector-button-2.png"
        />
        <img
          alt="vector-3"
          class="absolute left-[0px] top-[28px] w-[100px] rounded-bl-xl"
          src="/decorations/vector-button-3.png"
        />
        <div
          class="z-10 w-full text-start"
        >
          <p
            class="text-xs font-normal leading-[18px] text-white"
          >
            Baca Juga
          </p>
          <p
            class="text-sm font-bold leading-5 text-white"
          >
            Tujuan Program Indonesia Pintar
          </p>
        </div>
        <svg
          class="tabler-icon tabler-icon-chevron-right "
          fill="none"
          height="24"
          stroke="#ffffff"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 6l6 6l-6 6"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
